docs/development/benchmarking.md

```diff
@@ -0,0 +1,85 @@
+---
+sidebar_position: 3
+---
+
+# Benchmarking
+
+Do you have an existing Redis environment and would like to see if Dragonfly could be a better
+replacement? <br/>
+Are you developing a service and would like to determine which cloud instance type to
+allocate for Dragonfly? <br/>
+Do you wonder how many replicas you need to support your workload?
+
+If so, read on, because this page is for you!
+
+## Choosing an Environment
+
+A benchmark is done to assess the performance aspects of a system. In the case of Dragonfly, a
```

Comment:
Was this paragraph generated by ChatGPT? :)

I thought about providing specific requirements of how to say reach 1M qps on m5 family instance.
1. Specifiying Linux os version, even distribution, mentioning that Dragonfly runs best with iouring, 
2. talking about what's the minimal instance size that allows reaching 1M qps on m5.
3. Talking about running the load test on a separate machine. machine size for the load test .
4. Specifying that it is better to choose a bigger size for load test machine to avoid bottlenecks on the client side.
5. Provide rule of thumb for `--threads` should not be higher than number of vcpus on that machine. 
6. Provide suggested configuration for memtier to reach 1M qps on Dragonfly.
7. Provide suggested configuration + instance type to reach 2M qps on GCP instance etc

i.e. keep everything very technical and specific.

Reply:
or any other instance family and interesting target goal.

Reply:
It was not generated by ChatGPT, but I've been called worse :)

re/ 1: done
re/ 2: what is indeed the minimal instance?
re/ 3: I already talk about it below
re/ 4: done
re/ 5: done (but I think you meant `--proactor_threads`?)
re/ 6+7: do you have these? or would you like me to run them until I figure it out?

Reply:
Sorry 😞 , I was joking.

re/5 - I meant `-t` on memtier side. Dragonfly actually spans all the cpus automatically. memtier always uses 4 by default.